### PR TITLE
AlignTraj Writer kwargs fix

### LINKED ISSUE #4564: AlignTraj doesn't accept writer kwargs
## Expected behavior ##

When creating an aligned trajectory which is written to a file rather than stored in memory, I want to be able to pass in kwargs to the writer like I normally would when creating a writer object, but AlignTraj doesn't allow this.

## Actual behavior ##

AlignTraj's init method creates a writer for the provided filename like this:

``` python
self._writer = mda.Writer(self.filename, natoms)
```

## Code to reproduce the behavior ##

Example use case, when writing to h5md and you want the output file to be compressed.

``` python
from MDAnalysisTests.datafiles import COORDINATES_TOPOLOGY, COORDINATES_H5MD
from MDAnalysis.analysis import rms, align
import h5py

aligner = align.AlignTraj(u, ref,
                          select='protein and name CA',
                          filename='aligned_traj.h5md',
                          compression='gzip', compression_opts=9,
                          in_memory=False).run()
with h5py.File('aligned_traj.h5md', 'r') as f:
    print(f"{f["particles"]["trajectory"]["position"]["value"].compression}")
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
2.8.0

### PR DESCRIPTION
**Pull request by ljwoods2**
Fixes #4564

Changes made in this Pull Request:
 - Updated AlignTraj's writer constructor to take kwargs
 - Wrote test in test_align.py using xtc writer with "precision" kwarg

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4565.org.readthedocs.build/en/4565/

### CODE REVIEW

File: package/CHANGELOG [hunk 1]
```diff
@@ -37,6 +37,8 @@ Fixes
  * Fix doctest errors of lib/picklable_file_io.py (Issue #3925, PR #4371)
  * Fix deploy action to use the correct version of the pypi upload action.
  * Fix groups.py doctests using sphinx directives (Issue #3925, PR #4374)
+ * Fix `MDAnalysis.analysis.align.AlignTraj` not accepting writer kwargs
```
Comment (yuxuanzhuang) [inline package/CHANGELOG]: Newest first in the CHANGELOG.

File: package/MDAnalysis/analysis/align.py [hunk 2]
```diff
@@ -756,6 +756,7 @@ def __init__(self, mobile, reference, select='all', filename=None,
           for the remainder of the Python session. If ``mobile.trajectory`` is
           already a :class:`MemoryReader` then it is *always* treated as if
           ``in_memory`` had been set to ``True``.
+        - Writer ``**kwargs`` will be passed to the constructed `mda.Writer`.
```
Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/align.py]: use `writer_kwargs` dictionary instead to differentiate them from normal `kwargs`. Some writers cannot accept e.g. `verbose`.

**Review by yuxuanzhuang — CHANGES_REQUESTED** [review]

File: package/MDAnalysis/analysis/align.py (hunk 2 above)
Comment (yuxuanzhuang) [reply package/MDAnalysis/analysis/align.py]: @ljwoods2 Please leave review comments unresolved; reviewers may need to verify that the comments have indeed been addressed. :)

File: package/MDAnalysis/analysis/align.py [hunk 3]
```diff
@@ -720,6 +721,8 @@ def __init__(self, mobile, reference, select='all', filename=None,
         verbose : bool (optional)
              Set logger to show more information and show detailed progress of
              the calculation if set to ``True``; the default is ``False``.
+        writer_kwargs : dict (optional)
+            kwarg dict to be passed to the constructed writer
```
Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/align.py:725]: add a `.. versionchanged::` entry. See l675 as an example

**Review by yuxuanzhuang — APPROVED** [review]
Thanks so much! The functional parts all look good to me!

Regarding your final black modification, please see https://github.com/MDAnalysis/mdanalysis/issues/2450 for details but you should only format the lines you change to avoid complaints from https://github.com/MDAnalysis/mdanalysis/pull/4565#issuecomment-2052228126.

**Comment by ljwoods2** [comment]
My apologies, I forgot to turn off my autoformatter

**Review by hmacdope — APPROVED** [review]
LGTM thanks @ljwoods2

**Pull request merged**